### ISSUE TITLE
Use spec concept of Navigable

### ISSUE THREAD

**Issue by gterzian** [servo/servo#31520]
As I am looking into https://github.com/servo/servo/pull/31505, I came across what to me is a new concept in the spec: the [navigable](https://html.spec.whatwg.org/multipage/#infrastructure-for-sequences-of-documents:navigable). 

It is described in the spec as:

> a user-facing representation of a sequence of documents, i.e., they represent something that can be navigated between documents. Typical examples are tabs or windows in a web browser, or [iframe](https://html.spec.whatwg.org/multipage/iframe-embed-object.html#the-iframe-element)s, or [frame](https://html.spec.whatwg.org/multipage/obsolete.html#frame)s in a [frameset](https://html.spec.whatwg.org/multipage/obsolete.html#frameset).

This appears to match what we call a Webview, see https://github.com/servo/servo/blob/3a3e76a935f92ce84c24496cfc46207cd46088f6/components/constellation/webview.rs#L10

If that is correct, I think it would be nice to use the same name as the spec. 

Besides just renaming, I think we should try to stay as close to the spec as possible in terms of managing these, and their interactions with other concepts like session history, as well as the "update the rending" stuff in the HTML event-loop. As a quick example of how this matters, see https://html.spec.whatwg.org/multipage/webappapis.html#event-loop-processing-model:window-event-loop-3, which states that:

> An HTML event-loop(which in `script` is the `ScriptThread`, and in the constellation is an `EventLoop`) must wait for a rendering opportunity for an active document for one navigable. 

That sentence actually crosses between `script` and `constellation`, in the sense that "active document" involves the session history, and "navigable" could be a top-level BC or an iframe in that event-loop. Also "rendering opportunity" requires some knowledge of the rendering cycle. 

Needless to say, to properly "wait for a rendering opportunity for an active document for one navigable" will require some cooperation between script and the constellation beyond what we are currently doing(I think). 

For example, [running the resize steps](https://drafts.csswg.org/cssom-view/#document-run-the-resize-steps) for a document should be done as part of the "update the rendering" task in `script`, but currently we are simply [handling](https://github.com/servo/servo/blob/3a3e76a935f92ce84c24496cfc46207cd46088f6/components/script/script_thread.rs#L3787) incoming [messages from the constellation](https://github.com/servo/servo/blob/3a3e76a935f92ce84c24496cfc46207cd46088f6/components/script/script_thread.rs#L1576) for a given pipeline, which to me appears to be lacking the necessary context to determine whether this is an actual rendering opportunity for an active docume of (which?) navigable. 

I'll try to come-up with more specific examples as I work through https://github.com/servo/servo/pull/31505, and make this issue more actionable.

**Comment by gterzian** [comment]
@delan @mrobinson @wusyong @jdm

**Comment by gterzian** [comment]
> a WebView, which is a top level series of Documents

Sounds like https://html.spec.whatwg.org/multipage/#traversable-navigable

**Comment by mrobinson** [comment]
> > a WebView, which is a top level series of Documents
> 
> Sounds like https://html.spec.whatwg.org/multipage/#traversable-navigable (which is indeed a narrower concept).

I think it's a great idea to use the spec language the for code that implements it, but I'm not sure it makes sense for this to always be exposed in the Constellation and definitely not the API. Folks using Servo will know what a WebView is, which make it a very useful name.

**Comment by wusyong** [comment]
I agree WebView should be a term for external users. The code implementation itself should keep with spec.
I believe the doc comment of [Webivew](https://doc.servo.org/constellation/constellation/struct.WebView.html) already mentions it's a similar concept to a top-level browsing context.
